src/main/kotlin/com/cognifide/gradle/aem/common/instance/check/CheckRunner.kt

```diff
@@ -91,34 +91,44 @@ class CheckRunner(internal val aem: AemExtension) {
             val instance = progress.instance
             progress.stateWatch.start()
 
-            do {
-                if (aborted) {
-                    logger.info("Checking aborted for $instance")
-                    break
-                }
-
-                val checks = check(instance)
-                progress.currentCheck = checks
-
-                if (progress.stateChanged) {
-                    progress.stateChanges++
-                    progress.stateWatch.apply { reset(); start() }
-                }
-
-                progress.previousCheck = progress.currentCheck
+            logger.info("Checking started for $instance")
 
-                if (checks.done) {
-                    logger.info("Checking done for $instance")
-                    break
+            do {
+                try {
+                    if (aborted) {
+                        logger.info("Checking aborted for $instance")
+                        break
+                    }
+
+                    val checks = doChecking(progress)
+                    if (checks.done) {
+                        logger.info("Checking done for $instance")
+                        break
+                    }
+
+                    Behaviors.waitFor(delay.get())
+                } catch (e: Exception) {
```

Comment:
this is to catch corner case related to coroutine per instance which is sometimes crashing (very very rare case); this will prevent crashing coroutine and enables to be able to fix it as of now stack trace will be logged to the console.